targets/TARGET_NXP/TARGET_MCUXpresso_MCUS/TARGET_MIMXRT1050/device/TOOLCHAIN_ARM_STD/MIMXRT1052xxxxx.sct

```diff
@@ -138,14 +138,16 @@ LR_IROM1 m_flash_config_start m_text_start+m_text_size-m_flash_config_start {
   VECTOR_RAM m_interrupts_start EMPTY 0 {
   }
 #endif
-  RW_m_data m_data_start m_data_size-Stack_Size-Heap_Size { ; RW data
+  RW_m_data m_data_start m_data_size { ; RW data
     .ANY (+RW +ZI)
     *(m_usb_dma_init_data)
     *(m_usb_dma_noninit_data)
   }
-  RW_IRAM1 +0 EMPTY Heap_Size {    ; Heap region growing up
+  RW_IRAM1 ImageLimit(RW_m_data)  {
   }
-  ARM_LIB_STACK m_data_start+m_data_size EMPTY -Stack_Size { ; Stack region growing down
+  ARM_LIB_HEAP AlignExpr(+0, 16) EMPTY (m_data_start + m_data_size - Stack_Size - AlignExpr(ImageLimit(RW_IRAM1), 16)) {
```

Comment:
We don't need the ARM_LIB_HEAP section. RW_IRAM1 is used to define the HEAP space